src/Commands/LaravelRouteStatisticsCommand.php

```diff
@@ -29,11 +29,14 @@ class LaravelRouteStatisticsCommand extends Command
     public function handle()
     {
         $query = $this->getQuery();
+        $fields = $this->getFields();
 
         if ($this->option('group')) {
             $query->select($this->option('group'))
                 ->addSelect(DB::raw('MAX(date) as last_used'))
                 ->addSelect(DB::raw('SUM(counter) as counter'));
+        } else {
+            $query->select($fields);
```

Comment:
I get that this is in theory a small performance improvement, but it does break if you start adding fields that are not nessesarely DB fields but accessors.

```suggestion
```

Reply:
I didn't include it for the performance it was rather that I used a selector that was not in the table. I have a local change that splits up successors (e.g. for grouping) and real column names. In this change I also did some checks if other options where valid. Since it was out of scope of this PR I didn't add it. I will revert this to the original code.

Reply:
I think this is where you ment to post this @svenbw  :)

> As it turns out console table doesn't like an array, so I need to map each table element. And I now remember WHY the select is in place
> 
> ```
> $results = $query
>     ->limit($this->option('limit'))
>     ->get()
>     ->map(function(RouteStatistic $item) {
>         $data = $item->toArray();
>         if (config('route-statistics.store_parameters')) {
>             $data['parameters'] = json_encode($data['parameters']);
>         } else {
>             unset($data['parameters']); // Hide parameters if not supported in config
>         }
>         return $data;
>     });
> ```

Instead of only selecting the fields, then I suggest that we just pluck those fields AFTER query execution. That way it would work with accessors as well + solve your issue with arrays?

Reply:
Forgetting the `parameters` key by looping over the collection works if `parameters` is not enabled. However if they are still there it still needs to be casted to a string for table.

```php
$results = $query
    ->limit($this->option('limit'))
    ->get()
    ->map(function (RouteStatistic $item) {
        $data = $item->toArray();
        if (config('route-statistics.store_parameters')) {
            $data['parameters'] = json_encode($data['parameters']);
        } else {
            unset($data['parameters']);
        }
        return $data;
    });
```
or

```php
$results = $query
    ->limit($this->option('limit'))
    ->get()
    ->toArray();

if (config('route-statistics.store_parameters')) {
    Arr::map($results, function (array $item) {
        $item['parameters'] = json_encode($item['parameters']);
    });
} else {
    $results = Arr::map($results, function (array $item) {
        unset($item['parameters']);
        return $item;
    });
}
```

This is less efficient because php has to loop though each entry, else the data is collected by the database.

Can you explain why you would include other columns that are not in the fields? The table will be faulty because the headers of the table are generated by the `getFields()` function. If there is an additional column in the database the table will included empty headers and depending on the order even a field name above another column.

And @bilfeldt thanks for keeping the discussion going, I think we are getting there.

Reply:
Maybe something like:

```php
$results = $query
    ->limit($this->option('limit'))
    ->get()
    ->pluck($fields)
    ->map(function (array $data): array {
        return array_map(function ($item): string {
            if (is_array($item)) {
                return json_encode($item);
            }

            return (string) $data;
        }, $data);
    }) 
```

Reply:
pluck doesn't work, since it will only pluck items with specific keys from an array, and at that point there are still rows.
I filtered each row with `only` which will eliminate the `parameters` column.

I think the idea of `pluck` is the same as using `select`, where select has the benefit of offloading the complexity to the database, and thus slightly more performant.